atc/db/pipeline_lifecycle_test.go

```diff
@@ -10,70 +12,124 @@ import (
 		})
 
-		Context("parent pipeline is destroyed", func() {
+		Context("child pipeline is set by a job in a pipeline", func() {
+			var (
+				childPipeline db.Pipeline
+			)
+
 			BeforeEach(func() {
-				defaultPipeline.Destroy()
+				build, _ := defaultJob.CreateBuild()
+				childPipeline, _, _ = build.SavePipeline("child-pipeline", defaultTeam.ID(), defaultPipelineConfig, db.ConfigVersion(0), false)
+				build.Finish(db.BuildStatusSucceeded)
 			})
 
-			It("should archive all child pipelines", func() {
-				childPipeline.Reload()
-				Expect(childPipeline.Archived()).To(BeTrue())
-			})
-		})
+			Context("parent pipeline is destroyed", func() {
+				BeforeEach(func() {
+					defaultPipeline.Destroy()
+				})
 
-		Context("parent pipeline is archived", func() {
-			BeforeEach(func() {
-				defaultPipeline.Archive()
+				It("should archive all child pipelines", func() {
+					childPipeline.Reload()
+					Expect(childPipeline.Archived()).To(BeTrue())
+				})
 			})
 
-			It("should archive all child pipelines", func() {
-				childPipeline.Reload()
-				Expect(childPipeline.Archived()).To(BeTrue())
+			Context("parent pipeline is archived", func() {
+				BeforeEach(func() {
+					defaultPipeline.Archive()
+				})
+
+				It("should archive all child pipelines", func() {
+					childPipeline.Reload()
+					Expect(childPipeline.Archived()).To(BeTrue())
+				})
 			})
-		})
 
-		Context("job is removed from the parent pipeline", func() {
-			BeforeEach(func() {
-				defaultPipelineConfig.Jobs = atc.JobConfigs{
-					{
-						Name: "a-different-job",
-					},
-				}
-				defaultTeam.SavePipeline("default-pipeline", defaultPipelineConfig, defaultPipeline.ConfigVersion(), false)
+			Context("job is removed from the parent pipeline", func() {
+				BeforeEach(func() {
+					defaultPipelineConfig.Jobs = atc.JobConfigs{
+						{
+							Name: "a-different-job",
+						},
+					}
+					defaultTeam.SavePipeline("default-pipeline", defaultPipelineConfig, defaultPipeline.ConfigVersion(), false)
+				})
+
+				It("archives all child pipelines set by the deleted job", func() {
+					childPipeline.Reload()
+					Expect(childPipeline.Archived()).To(BeTrue())
+				})
 			})
+		})
 
-			It("archives all child pipelines set by the deleted job", func() {
-				childPipeline.Reload()
-				Expect(childPipeline.Archived()).To(BeTrue())
+		Context("pipeline does not have a parent job and build ID", func() {
+			It("Should not archive the pipeline", func() {
+				defaultPipeline.Reload()
+				Expect(defaultPipeline.Archived()).To(BeFalse())
 			})
 		})
 	})
 
-	Context("pipeline does not have a parent job and build ID", func() {
-		It("Should not archive the pipeline", func() {
-			defaultPipeline.Reload()
-			Expect(defaultPipeline.Archived()).To(BeFalse())
+	Describe("RemoveBuildEventsForDeletedPipelines", func() {
+		var (
+			pipeline1 db.Pipeline
+			pipeline2 db.Pipeline
+		)
+
+		BeforeEach(func() {
+			pipeline1 = defaultPipeline
```

Comment:
I was confused why we didn't save `pipeline1` until I realized `defaultPipeline` was saved in the suite setup. Maybe we should use it directly instead of renaming it (e.g. `defaultPipeline.Destroy()` down below)

Reply:
Fair, the asymmetric setup is a little weird. I'd rather keep the consistent naming of `pipeline1`/`pipeline2` though, rather than `defaultPipeline`/`pipeline2`, so I'll just save a new pipeline for `pipeline1` as well